src/auth0/handlers/rules.js

```diff
@@ -87,7 +87,7 @@ export default class RulesHandler extends DefaultHandler {
 
     // Figure out the rules that need to be re-ordered
     const futureRules = [ ...create, ...update ];
-    let nextOrderNo = Math.max(...existing.map(r => r.order)) + 1;
+    let nextOrderNo = Math.max(Math.max(...futureRules.map(r => r.order)), Math.max(...existing.map(r => r.order)));
```

Comment:
This is a pretty dense line. It might be a little easier to read if we move the future rules array of orders and the existing rule orders to declared variables above.

Reply:
Also, will this work recursively? If I have two `[1,2]` and `[2,3]`, will  *both* of the rules in the second set be reordered to `[3,4]`?

It might be worth adding a test case for this scenario.